src/docfx/serve/lsp/DidChangeWatchedFilesHandler.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Microsoft.Docs.Build
+{
+    internal class DidChangeWatchedFilesHandler : IDidChangeWatchedFilesHandler
+    {
+        private readonly LanguageServerBuilder _languageServerBuilder;
+        private readonly ILanguageServerNotificationListener _notificationListener;
+        private readonly LanguageServerPackage _package;
+
+        private readonly Container<FileSystemWatcher> _watcher = new Container<FileSystemWatcher>(
+                new FileSystemWatcher() { GlobPattern = "**/*", Kind = WatchKind.Create | WatchKind.Delete | WatchKind.Change });
+
+        public DidChangeWatchedFilesHandler(
+            LanguageServerBuilder languageServerBuilder,
+            ILanguageServerNotificationListener notificationListener,
+            LanguageServerPackage package)
+        {
+            _languageServerBuilder = languageServerBuilder;
+            _notificationListener = notificationListener;
+            _package = package;
+        }
+
+        public Task<Unit> Handle(DidChangeWatchedFilesParams notification, CancellationToken cancellationToken)
+        {
+            var hasInScopedFileChange = notification.Changes.Any(
```

Comment:
The watcher is configured to watch `**/*`, if that means only watch files under the current working directory (or we could instruct the watcher to do so), will we still hit a case where a file changed outside of the working directory causes a change event here?

Reply:
It means only watch files under  the current working directory, but it will get the notifications for the files under `C:\Users\{user}\AppData\Roaming\Code\User\globalStorage` folder

Reply:
That's a wired behavior, what if the glob pattern is configured to be `{working-directory}/**`?

Reply:
remove the code to detect `hasInScopedFileChange`?